src/io/parquet/read/deserialize/simple.rs

```diff
@@ -401,17 +439,14 @@ fn timestamp<'a, I: Pages + 'a>(
     time_unit: TimeUnit,
 ) -> Result<ArrayIter<'a>> {
     if physical_type == &PhysicalType::Int96 {
-        let iter = primitive::Iter::new(pages, data_type, num_rows, chunk_size, int96_to_i64_ns);
-        let logical_type = PrimitiveLogicalType::Timestamp {
-            unit: ParquetTimeUnit::Nanoseconds,
-            is_adjusted_to_utc: false,
-        };
-        let (factor, is_multiplier) = unifiy_timestmap_unit(&Some(logical_type), time_unit);
-        return match (factor, is_multiplier) {
-            (1, _) => Ok(dyn_iter(iden(iter))),
-            (a, true) => Ok(dyn_iter(op(iter, move |x| x * a))),
-            (a, false) => Ok(dyn_iter(op(iter, move |x| x / a))),
+        let conversion_op = match time_unit {
+            TimeUnit::Nanosecond => int96_to_i64_ns,
+            TimeUnit::Microsecond => int96_to_i64_us,
+            TimeUnit::Millisecond => int96_to_i64_ms,
+            TimeUnit::Second => int96_to_i64_s,
         };
+        let iter = primitive::Iter::new(pages, data_type, num_rows, chunk_size, conversion_op);
```

Comment:
I think we should branch this iterator on the time unit. Otherwise the inline doesn't do much as it is still runtime dependable.

Can you move this iter creation and return in the match case?